web_pivot_zoom/static/src/js/web_pivot_zoom.js

```diff
@@ -0,0 +1,205 @@
+                function(field)
+                {
+                    return self.pivot_options.invisible_measures.indexOf(
+                        field.field
+                    ) == -1;
+                }
+            );
+        },
+        _web_pivot_zoom_on_click_cell: function(e)
+        {
+            if(!e.currentTarget.cellIndex)
+            {
+                return;
+            }
+            var self = this,
+                $current_row = jQuery(e.currentTarget).closest('tr'),
+                $col_row = jQuery(e.currentTarget).closest('table')
+                .find('thead tr:nth-last-child(2)'),
+                col_id = $col_row.find('[data-id]').eq(
+                    Math.floor(
+                        (e.currentTarget.cellIndex - 1) /
+                        this.pivot.measures.length
+                    )
+                ).attr('data-id'),
+                row_id = $current_row.find('[data-id]').attr('data-id'),
+                current_field = this.pivot.measures[
+                    (e.currentTarget.cellIndex - 1) %
+                    this.pivot.measures.length
+                ],
+                row_header = this.pivot.get_header(row_id),
+                col_header = this.pivot.get_header(col_id),
+                measure_values = this.pivot.get_values(
+                    Math.min(col_id, row_id), Math.max(col_id, row_id)
+                ),
+                group_values = {};
+            _.each(this.groupby_fields, function(field)
+            {
+                group_values[field.field] = null;
+            });
+            _.each(row_header.domain, function(leaf)
+            {
+                group_values[leaf[0]] = leaf[2];
+            });
+            _.each(col_header.domain, function(leaf)
+            {
+                group_values[leaf[0]] = leaf[2];
+            });
+            _.each(this.pivot_options.measures, function(measure, i)
+            {
+                group_values[measure.field] = measure_values[i];
+            });
+            _.each(this.pivot_options.invisible_measures, function(measure, i)
+            {
+                group_values[measure] = measure_values[
+                    i + self.pivot_options.measures.length
+                ];
+            });
+            var model = this.graph_view._web_pivot_zoom[current_field.field]
+                .model,
+                domain = instance.web.pyeval.eval(
+                    'domain',
+                    this.graph_view._web_pivot_zoom[current_field.field]
+                    .domain || [],
+                    group_values
+                );
+            if(!model || !domain)
+            {
+                return;
+            }
+            return this._web_pivot_zoom_action(
+                model, domain, current_field, row_header, col_header
+            );
+        },
+        _web_pivot_zoom_action: function(
+            model, domain, current_field, row_header, col_header
```

Comment:
this is a precaution for the future - we might need it later, and then it would be a pity to have to change the api. I usually try to make my code extensible by default